### PR TITLE
Add Swiss number separator style

### LINKED ISSUE #13348: Add Swiss number formatting (Localization - Separator Style)
🤚 _I'm happy to contribute, but I need some pointers (links) where to find the relevant files for l10n / i18n._

**Is your feature request related to a problem? Please describe.**
The Swiss number format is not supported currently. We operate in Switzerland and our numbers on the dashboards are never quite what we expect (pun intended).

Note: even though the examples below include currency symbols, this issue concerns the group and decimal separators.

**Describe the solution you'd like**
Add a new _Separator Style_ in Settings > Localization > Numbers. That's `<Site URL>/admin/settings/localization`.

The Swiss separator style follows this pattern `100’000’000.00`.

**Describe alternatives you've considered**
I'm not familiar with the Metabase architecture, but I assume that there is a frontend part and a backend part to it. First priority would be to address the frontend.

**How important is this feature to you?**
I created an issue and would like to contribute this myself. That important. Help appreciated.

**Additional context**
Example of the Swiss format compared to German and US, taken from [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/NumberFormat):
```js
const number = 123456.789;

console.log(new Intl.NumberFormat('de-CH', { style: 'currency', currency: 'CHF' }).format(number));
console.log(new Intl.NumberFormat('de-DE', { style: 'currency', currency: 'EUR' }).format(number));
console.log(new Intl.NumberFormat('en-US', { style: 'currency', currency: 'USD' }).format(number));

> "CHF 123’456.79"
> "123.456,79 €"
> "$123,456.79"
```

This looks like a good place to look into for the frontend part.
- https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/formatting.js#L52
- https://github.com/metabase/metabase/blob/master/frontend/src/metabase/lib/formatting.js#L230-L239

### PR DESCRIPTION
**Pull request by etoulas**
The Swiss separator style follows this pattern 100’000’000.00

### Tests
-  [x] Run the frontend and integration tests with  `yarn lint && yarn flow && yarn test`)

-  [x] Sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform)
(unless it's a tiny documentation change).

Resolves #13348

### CODE REVIEW

**Comment by etoulas** [comment]
The obvious place to add the number formatting in the frontend didn't work out at all.
I went to /admin/settings/localization and reviewed the _Separator style_ drop-down. But it still contains the same 4 entries as before.

Different things that I might be doing wrong:
- Wrong use of yarn / frontend build process.
- There are more/different places that need to be adjusted.
- Or else?

**Comment by camsaul** [comment]
@paulrosenzweig do you know what other places @etoulas would need to adjust to add a new number separator?

**Comment by flamber** [comment]
I'm sure you need to save the setting here too https://github.com/metabase/metabase/blob/master/frontend/src/metabase/visualizations/lib/settings/column.js#L379-L391 - Haven't tested, and there might be other places, but that should get you started.

**Comment by etoulas** [comment]
Thanks for the hint @flamber, appreciate it! That did the trick. This time I also grepped the whole frontend code base for number/separators keywords but didn't find any more places.

The tests only cover the default number separator https://github.com/metabase/metabase/blob/master/frontend/test/metabase/lib/formatting.unit.spec.js#L23-L27 - I couldn't figure out how to set up the test with a different separator in order to add additional test cases.

**Comment by paulrosenzweig** [comment]
@etoulas sorry we don't already have test coverage of the existing separator options!

You could call `formatNumber` with the `number_separators` option set. The logic is the same regardless of what the separators are so we probably don't need to test each separator separately. 

`expect(formatNumber(1000.1, { number_separators: ".’" })).toEqual("1'000.1");`

**Review by paulrosenzweig — APPROVED** [review]
LGTM! Thanks for the PR!

**Comment by etoulas** [comment]
My pleasure @paulrosenzweig. Thanks for your support!

**Comment by etoulas** [comment]
I thought about adding another cypress test along those lines https://github.com/metabase/metabase/blob/master/frontend/test/metabase/scenarios/admin/settings/settings.cy.spec.js#L156-L177
But time!

File: frontend/src/metabase/lib/formatting.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ import type {
 import type { ClickObject } from "metabase-types/types/Visualization";
 
 // a one or two character string specifying the decimal and grouping separator characters
-export type NumberSeparators = ".," | ", " | ",." | ".";
+export type NumberSeparators = ".," | ", " | ",." | "." | "’.";
```
Comment (paulrosenzweig) [inline frontend/src/metabase/lib/formatting.js:57]: @etoulas I think the `'` and `.` backwards here

Comment (etoulas) [reply frontend/src/metabase/lib/formatting.js:57]: You are right! I wonder where this is used, because I couldn't spot the difference.

Comment (paulrosenzweig) [reply frontend/src/metabase/lib/formatting.js]: They types are only used in our flow type checking. They don't make it into the compiled app code. It looks like the flow check passed, so I guess that type isn't used effectively enough. Still good to keep it up to date though.

**Pull request merged**